### PR TITLE
Rename tgamma to gamma

### PR DESCRIPTION
**Pull request by Paalon**
Rename `tgamma` (means true gamma) to `gamma`. Deprecate `tgamma`.
http://pubs.opengroup.org/onlinepubs/009695399/functions/tgamma.html
*This function is named tgamma() in order to avoid conflicts with the historical gamma() and lgamma() functions.*
There are no need to use the name `tgamma` in Nim.

### CODE REVIEW

File: lib/pure/math.nim [hunk 1]
```diff
@@ -274,12 +274,16 @@ when not defined(JS): # C
   proc erfc*(x: float64): float64 {.importc: "erfc", header: "<math.h>".}
     ## The complementary error function
 
+  proc gamma*(x: float32): float32 {.importc: "tgammaf", header: "<math.h>".}
+  proc gamma*(x: float64): float64 {.importc: "tgamma", header: "<math.h>".}
+    ## The gamma function
+  proc tgamma*(x: float32): float32 {.deprecated, importc: "tgammaf", header: "<math.h>".}
+  proc tgamma*(x: float64): float64 {.deprecated, importc: "tgamma", header: "<math.h>".}
+    ## The gamma function
+    ## **Deprecated since version x.xx.x**: Use ``gamma`` instead.
```
Comment (dom96) [inline lib/pure/math.nim]: Just write "0.19.0".

**Comment by dom96** [comment]
What are these "historical" functions, and why are they historical?

**Comment by Paalon** [comment]
Some C compilers had used `gamma` as log gamma and others as gamma. In C99 `tgamma` introduced as gamma function for compatibility.
https://www.gnu.org/software/libc/manual/html_node/Special-Functions.html

**Comment by Araq** [comment]
This also needs a changelog entry. Also consider to use the newer ``.deprecated: "use gamma instead"`` syntax.

**Pull request merged**